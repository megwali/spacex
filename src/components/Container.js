import { Route, Switch } from 'react-router-dom';
import { Dashboard, Details } from '../pages';


const Container = () => (
  <Switch>
    <Route exact path="/">
      <Dashboard />
    </Route>

    <Route path="/:id">
      <Details />
    </Route>
  </Switch>
);

export default Container;
