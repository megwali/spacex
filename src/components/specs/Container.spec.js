import { create } from 'react-test-renderer';
import Container from '../Container';

jest.mock('react-router-dom', () => ({
  Link: 'Link',
  Route: 'Route',
  Switch: 'Switch',
  useLocation: () => ({ pathname: '/1' }),
}));
jest.mock('../../hooks/useLaunchList', () => () => ({
  launches: [{ flight_number: 1, launch_date_local: '2020-08-30T19:18:00-04:00' }]
}));
jest.mock('../../pages', () => ({ Dashboard: 'Dashboard', Details: 'Details' }));

describe('Container', () => {
  it ('renders', () => {
    const component = create(<Container />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
