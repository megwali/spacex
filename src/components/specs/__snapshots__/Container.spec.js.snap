// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
  >
    <div
      className="sc-pFZIQ kabEES"
    >
      <Link
        className="sc-bdfBwQ CdKhw"
        to="/1"
      >
        <div
          className="sc-gsTCUz fOxzTe"
        >
          <img
            height="56"
            src="spacex_logo_square.png"
            width="56"
          />
          <div
            className="sc-hKgILt cVPoMr"
          >
            <p
              className="sc-eCssSg liPRvx"
            />
            <p
              className="sc-dlfnbm NlWAF"
            >
              Aug 31st 2020, 12:18:00 am
            </p>
          </div>
        </div>
        <div>
          <p
            className="sc-jSgupP sc-gKsewC bPHQOa eOGKIx"
          >
            Failed
          </p>
        </div>
      </Link>
    </div>
  </Route>
  <Route
    path="/:id"
  >
    <div
      className="sc-jrAGrp ktGWEx"
    >
      <p
        className="sc-iqHYGH ghVZyH"
      >
        Rocket name: 
      </p>
      <p
        className="sc-kEjbxe bipcXF"
      >
        Rocket type: 
      </p>
      <p
        className="sc-kEjbxe bipcXF"
      >
        Reused: 
        No
      </p>
    </div>
  </Route>
</Switch>
`;
